ProfileManager.js

```diff
@@ -136,9 +144,18 @@ export default class ProfileManager {
     const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
 
     const capabilityCache = this._getCache(capabilityCacheKey);
-    const agentZcap = capabilityCache.get(capabilityKey);
+    const agentZcap = await capabilityCache.get(capabilityKey);
+    const now = Date.now();
     if(agentZcap) {
-      return agentZcap;
+      if(!agentZcap.expires) {
+        return agentZcap;
+      }
+      const expiryDate = new Date(agentZcap.expires);
+      const timeDiff = expiryDate.getTime() - now;
+      if(timeDiff > this.gracePeriod) {
+        return agentZcap;
+      }
+      capabilityCache.del(capabilityKey);
     }
```

Comment:
Given that we're awaiting the result of the cache above ... I think we have to be more careful here. We should only delete what's in the `capabilityCache` if it matches what we got out of it. So, we'd have to get the promise from the cache and store that in a variable ... then await that promise to get the `agentZcap` -- and then check that promise instance against what's in the cache before deleting. Otherwise, another process may have already updated the `capabilityCache` with the appropriate value.

All of that sounds messy ... so luckily I think there's a better fix, which is to just set the `maxAge` for the entry when we put it in the cache. Then we can also avoid checking the other stuff here and get a simpler implementation.

So, when we do `capabilityCache.set` ... we should be doing:

```js
const maxAge = /* math to figure out capability.expires - zcapGracePeriod */
capabilityCache.set(capabilityKey, promise, maxAge);
```

Then, we don't have to worry about checking the zcap when we pull it out of the cache -- the lru-cache will do it for us. If we get a result, it's not expired. Make sense? We can do the same for the agent signer.